programs/locker/src/instructions/v3/create_vesting_escrow_metadata.rs

```diff
@@ -0,0 +1,42 @@
+use crate::*;
+
+/// Accounts for [locker::create_vesting_escrow_metadata].
+#[derive(Accounts)]
+#[instruction(metadata: CreateVestingEscrowMetadataParameters)]
+pub struct CreateVestingEscrowMetadataV3<'info> {
+    /// The [Escrow].
+    #[account(mut, has_one = creator)]
+    pub escrow: AccountLoader<'info, VestingEscrowV3>,
+    /// Creator of the escrow.
+    pub creator: Signer<'info>,
+    /// The [ProposalMeta].
+    #[account(
+        init,
+        seeds = [
+            b"escrow_metadata".as_ref(),
+            escrow.key().as_ref()
+        ],
+        bump,
+        payer = payer,
+        space = 8 + VestingEscrowMetadata::space(&metadata)
+    )]
+    pub escrow_metadata: Box<Account<'info, VestingEscrowMetadata>>,
+    /// Payer of the [ProposalMeta].
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    /// system program.
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handle_create_vesting_escrow_metadata_v3(
+    ctx: Context<CreateVestingEscrowMetadataV3>,
+    params: &CreateVestingEscrowMetadataParameters,
+) -> Result<()> {
+    let escrow_metadata = &mut ctx.accounts.escrow_metadata;
+    escrow_metadata.escrow = ctx.accounts.escrow.key();
+    escrow_metadata.name = params.name.clone();
+    escrow_metadata.description = params.description.clone();
+    escrow_metadata.creator_email = params.creator_email.clone();
+    escrow_metadata.recipient_email = params.recipient_email.clone();
```

Comment:
We dont need recipient email here because there are many recipients in escrow v3

Reply:
To reuse `VestingEscrowMetadata` in this case. Should we use empty string (`recipient_email = ""`) ?

Reply:
can, you can validate recipient_email == ""

Reply:
In vesting_escrow_metadata, we don't need recipient_email, but we need an endpoint for merkle_tree_api, that also be string, could you combine `recipient_email ` and `merkle_tree_api` in a common field, so we can make the endpoint work for both escrow v1 and escrow v3?

Reply:
i have updated `recipient_endpoint` field to make it common for both v1 & v2.

Reply:
I have removed `create_escrow_metadata_v3` and make common function create metadata for both v1 & v2